presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergTableChangelog.java

```diff
@@ -0,0 +1,200 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg;
+
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.iceberg.IcebergQueryRunner.createIcebergQueryRunner;
+
+public class TestIcebergTableChangelog
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Map<String, String> properties = ImmutableMap.of("http-server.http.port", "8080");
+        return createIcebergQueryRunner(properties, ImmutableMap.of());
+    }
+
+    private long[] snapshots = new long[0];
+
+    @Override
+    @BeforeClass
+    public void init()
+            throws Exception
+    {
+        super.init();
+        assertQuerySucceeds("CREATE TABLE ctas_orders as SELECT * FROM orders LIMIT 10");
+        assertQuerySucceeds("INSERT INTO ctas_orders SELECT * FROM orders LIMIT 20");
+        assertQuerySucceeds("INSERT INTO ctas_orders SELECT * FROM orders LIMIT 30");
+        snapshots = Lists.reverse(
+                        getQueryRunner().execute("SELECT snapshot_id FROM \"ctas_orders$snapshots\" ORDER BY committed_at").getOnlyColumn()
+                                .collect(Collectors.toList()))
+                // reverse and skip the latest snapshot ID since it's invalid
+                // to get the changelog for the current snapshot
+                .stream().skip(1)
+                .mapToLong(Long.class::cast)
+                .toArray();
+    }
+
+    @Test
+    public void testSchema()
+    {
+        assertQuery(String.format("SHOW COLUMNS FROM \"ctas_orders@%d$changelog\"", snapshots[0]),
+                "VALUES" +
+                        "('operation', 'varchar', '', '')," +
+                        "('ordinal', 'bigint', '', '')," +
+                        "('snapshotid', 'bigint', '', '')," +
+                        "('rowdata', 'row(\"orderkey\" bigint, \"custkey\" bigint, \"orderstatus\" varchar, \"totalprice\" double, \"orderdate\" date, \"orderpriority\" varchar, \"clerk\" varchar, \"shippriority\" integer, \"comment\" varchar)', '', '')");
+    }
+
+    @Test
+    public void testBasicSelect()
+    {
+        for (long id : snapshots) {
+            assertQuerySucceeds(String.format("SELECT * FROM \"ctas_orders@%d$changelog\"", id));
```

Comment:
In this file it would also be good to validate results in addition to checking whether the query succeeds.
For e.g. select a subset of columns from the changelog table (I believe only snapshotid is non-deterministic?) and use assertQuery where possible